fuzz/fuzz.go

```diff
@@ -0,0 +1,116 @@
+package fuzz
+
+import (
+	"bufio"
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"time"
+)
+
+var failingInputRegex = regexp.MustCompile(`^\s*go test -run=Fuzz([a-zA-Z0-9_]+)/([a-zA-Z0-9#]+)`)
+var failingSeedInputRegex = regexp.MustCompile(`^\s*failure while testing seed corpus entry: Fuzz([a-zA-Z0-9_]+)/([a-zA-Z0-9#]+)`)
+
+type FailingInputError struct {
+	ID   string
+	File string
+	Seed bool
+}
+
+func (f FailingInputError) Error() string {
+	newOrSeed := "new"
+	if f.Seed {
+		newOrSeed = "seed"
+	}
+
+	if f.File != "" {
+		return fmt.Sprintf("failing %s input, saved at %s", newOrSeed, f.File)
+	}
+	return fmt.Sprintf("failing %s input: %s", newOrSeed, f.ID)
+}
+
+func (p *Project) relCorpusDir(target TestTarget) string {
+	// target.Package contains the root package as well
+	// we need to strip it because it refers to the current working directory .
+	pkg := strings.TrimPrefix(strings.TrimPrefix(target.Package, target.RootPackage), "/")
+	return filepath.Join(pkg, "testdata/fuzz", target.Name)
+}
+
+func (p *Project) Fuzz(target TestTarget, d time.Duration) error {
+	args := []string{
+		"test",
+		"-test.run=^$",
+		"-test.fuzz=^" + target.Name + "$",
+		"-test.fuzztime=" + d.String(),
+		target.Package,
+	}
+
+	cmd := exec.Command("go", args...)
```

Comment:
Here we could use [exec.LookPath](https://pkg.go.dev/os/exec#LookPath), in order to bail if underlying platform does not have the go in PATH or installed. 

(Also from that command we can then use the actual bin path, instead of just `go`. Might be useful for Windows.)

Reply:
I'm not sure if this is required for Windows, when you look at the source code of [cmd.Exec](https://cs.opensource.google/go/go/+/refs/tags/go1.21.4:src/os/exec/exec.go;l=418-429) it already calls `LookPath` underneath.
I added `LookPath` for a more user-friendly error message if go is not installed.